### PR TITLE
Genericify refactor

### PR DESCRIPTION
**Pull request by jsbnr**
DO NOT MERGE NOT COMPLETE!

This PR contains a refactor of some of the configuration and functionality moving it from hard coded config/javascript to config options with the intention to make it easier to deploy and use across multiple dashboards.

Main changes include:

- Query is now configurable as a viz option 
- Disable zoom cluserting now a config option
- Center point of map can be specified as lat/lng coords, removed list
- Some naming updates to be less 'store' specific
- Changed how alert thresholds work, now powered by values defined in query (threshold_critical, threshold_warning), can also now work in both directions as required
- Removed hard coded required query result values and replaced with conventions (value, name, etc)
- Changed tooltip to appear on hover instead of click
- Made the tooltip content configurable via query values  named convention tooltip_xxx
- Changed cluster colour to neutral grey instead of green.
- Made marker displayed value configurable via nrql value (icon_label)
- Added config option for refresh interval
- Changed the empty state to be more useful
- Fixed zero location result causing oops
- Added option to provide link in NRQL results which can be navigated by clicking on a marker

TODO: 

- [x] Review with Mat
- [x] Decide what to do with regions

### CODE REVIEW

File: visualizations/store-map-viz/components/EmptyState.tsx [hunk 1]
```diff
@@ -4,19 +4,36 @@ import { Card, CardBody, HeadingText } from "nr1";
...
-      </HeadingText>
-      <code>FROM NrUsage SELECT sum(usage) FACET metric SINCE 1 week ago</code>
+      <div className="queryHelpText">
+      The query should include the columns (as required):<br /><br />
+        <ul>
+          <li>latitude - the latitude of the marker</li>
+          <li>longitude - the longitude of the marker</li>
+          <li>icon_label - the value to show on the marker</li>
+          <li>value - the value to evaluate against thresholds</li>
+          <li>threshold_warning - the threshold to cause warning colour</li>
```
Comment (matewilk) [inline visualizations/store-map-viz/components/EmptyState.tsx]: Add to READEME.md file how you can use `threshold_*` property by setting it in the query, either "hardcoded" in the query, coming from a lookup table etc.

File: visualizations/store-map-viz/constants.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+export const DEFAULT_ZOOM = 1; // default zoom level
+export const DEFAULT_CENTER = [51.5074, 0.1278]; // default map center - London
+export const FETCH_INTERVAL_DEFAULT = 300; // fetch interval in ms - 5 minutes
+export const MARKER_COLOURS = {
+    criticalColour : "#B22222", // Red color for alert
+    warningColour: "#DAA520", // Amber color for warning
+    safeColour : "#228B22", // Green color for safe
+    groupColor: "#999", //grey for group
+    groupBorder: "#9999994d" //border, including transparency
```
Comment (matewilk) [inline visualizations/store-map-viz/constants.ts]: Can we change the colours to align with NR theme colours?

File: visualizations/store-map-viz/components/LocationPopup.tsx [hunk 3]
```diff
@@ -1,24 +1,21 @@
...
+      return <>
+        <div className="popup-label">{item.label}:</div>
+        <div>{(typeof item.formatFn) === "function" ? item.formatFn?.(location[item.queryField]): location[item.queryField]}</div>
+      </>
+  })
 
-        <div className="popup-label">Amount:</div>
-        <div>{formatCurrency(location.amount)}</div>
+  return (
+    <Tooltip>
```
Comment (matewilk) [inline visualizations/store-map-viz/components/LocationPopup.tsx:19]: Add features of Tooltip (for example http link) to README.md file.

File: visualizations/store-map-viz/styles.scss [hunk 4]
```diff
@@ -56,14 +66,9 @@
...
-  background-color: rgba(
-    60,
-    179,
-    113,
-    0.8
-  ); /* Slightly darker green with some transparency */
-  box-shadow: 0 0 0 8px rgba(110, 204, 57, 0.3); /* Creates the outer ring */
+  color: #ffffff; /* Text color */
+  background-color:inherit; /* Slightly darker green with some transparency */
+  // box-shadow: 0 0 0 8px #cc39c04d; /* Creates the outer ring */
```
Comment (matewilk) [inline visualizations/store-map-viz/styles.scss]: Can we change the colours to NR colours here as well please?

**Review by matewilk — CHANGES_REQUESTED** [review]
Looks amazing, just a few minor comments.

File: visualizations/store-map-viz/constants.ts (hunk 2 above)
Comment (jsbnr) [reply visualizations/store-map-viz/constants.ts]: Done, used billboard colors.

File: visualizations/store-map-viz/styles.scss (hunk 4 above)
Comment (jsbnr) [reply visualizations/store-map-viz/styles.scss]: these are no longer used

**Comment by jsbnr** [comment]
Decision: to remove regions from the app, these can be added as a new feature so the code has been left in place but is not used.

File: visualizations/store-map-viz/queries/index.tsx [hunk 5]
```diff
@@ -1,43 +1,34 @@
...
-//       actor {
-//         account(id: $id) {
-//           lookup: ${lookupTableNrql()}
-//         }
-//       }
-//     }
-// `;
+    // Construct the full NRQL query, remove line breaks
+    let query = `${markersQuery.replace(/(\r\n|\n|\r)/gm," ")} ${timeRangePart}`;
+    console.log("OVERFETCH!",Date.now(), query); //TODO: Mat to look at overfetch on window resize
```
Comment (jsbnr) [inline visualizations/store-map-viz/queries/index.tsx]: @matewilk Can you look into this overfetch issue when resizing window?

Comment (matewilk) [reply visualizations/store-map-viz/queries/index.tsx]: I'll look into it

**Review by matewilk — COMMENTED** [review]
Looks good!

**Pull request merged**